lib/sapic/src/Sapic/Compression.hs

```diff
@@ -59,15 +59,17 @@ getProducedFacts rules = facts
 
 mergeInfo :: ProtoRuleEInfo -> ProtoRuleEInfo -> ProtoRuleEInfo
 mergeInfo (ProtoRuleEInfo (StandRule name) attr res) (ProtoRuleEInfo (StandRule name2) attr2 res2) =
-  ProtoRuleEInfo (StandRule (mergeStand name name2)) (mergeAttr attr attr2) (res ++ res2)
+  ProtoRuleEInfo (StandRule (mergeStand name name2)) (mergeAttrs attr attr2) (res ++ res2)
  where
        mergeStand n _ = n  -- ++ "_" ++ n'
        -- NOTE: concatenating makes veryyyy big name rules, that completely make the the graphs unreadble
        -- NOTE: if we reintroduce Yavor's Dot output, recall 9e7e99fe070776172bd09cb977e8d3a83da3ed51
-       mergeAttr a a' =  let completeList = a ++ a' in
+       mergeAttrs a a' =  let completeList = a ++ a' in
                             take 1 [i |  i@(RuleColor _) <- completeList]
                          ++ take 1 [i |  i@(Process   _) <- completeList]
                          ++ take 1 [i |  i@IsSAPiCRule   <- completeList]
+                         ++ take 1 [i |  i@IgnoreDerivChecks   <- completeList]
+                         ++ take 1 [i |  i@(Role _)   <- completeList]
```

Comment:
I've spent *hours* today finding this issue: attribute merging forgets the Role attribute when two rules are compressed. This is a hotfix, but this nasty problem reappears as soon as another attribute is introduced. We should not merge before we find a solution to this.

To the Haskell enthusiasts: what is the right way to solve this?